server/src/main/java/org/opensearch/action/admin/cluster/shards/routing/weighted/delete/ClusterDeleteWeightedRoutingRequest.java

```diff
@@ -8,37 +8,142 @@
 
 package org.opensearch.action.admin.cluster.shards.routing.weighted.delete;
 
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.OpenSearchGenerationException;
+import org.opensearch.OpenSearchParseException;
 import org.opensearch.action.ActionRequestValidationException;
 import org.opensearch.action.support.clustermanager.ClusterManagerNodeRequest;
+import org.opensearch.cluster.metadata.WeightedRoutingMetadata;
+import org.opensearch.common.bytes.BytesReference;
 import org.opensearch.common.io.stream.StreamInput;
 import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.xcontent.DeprecationHandler;
+import org.opensearch.common.xcontent.NamedXContentRegistry;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.common.xcontent.XContentFactory;
+import org.opensearch.common.xcontent.XContentHelper;
+import org.opensearch.common.xcontent.XContentParser;
+import org.opensearch.common.xcontent.XContentType;
 
 import java.io.IOException;
+import java.util.Map;
 
 /**
  * Request to delete weights for weighted round-robin shard routing policy.
  *
  * @opensearch.internal
  */
 public class ClusterDeleteWeightedRoutingRequest extends ClusterManagerNodeRequest<ClusterDeleteWeightedRoutingRequest> {
-    public ClusterDeleteWeightedRoutingRequest() {}
+    private static final Logger logger = LogManager.getLogger(ClusterDeleteWeightedRoutingRequest.class);
+
+    private long version;
+    private String awarenessAttribute;
+
+    public void setVersion(long version) {
+        this.version = version;
+    }
+
+    public ClusterDeleteWeightedRoutingRequest() {
+        this.version = WeightedRoutingMetadata.VERSION_UNSET_VALUE;
```

Comment:
Why does delete need to have an explicit version ?

Reply:
Basically set this value for testing otherwise version would get set to 0 (default value) if not set explicitly.

Reply:
Can we make this package private if used only for testing